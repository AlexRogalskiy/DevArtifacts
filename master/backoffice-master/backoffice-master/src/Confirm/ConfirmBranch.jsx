import React from 'react'
import PropTypes from 'prop-types'
import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
  withStyles,
} from '@material-ui/core'

const styles = theme => ({
  primaryButton: {
    color: theme.palette.primary.dark,
  },
})

const ConfirmBranch = ({
  title,
  description,
  isOpen,
  disagreeText,
  agreeText,
  hasCloseButton,
  onClose,
  onConfirm,
  classes,
}) => (
  <Dialog
    open={isOpen}
    onClose={onClose}
    aria-labelledby="alert-dialog-title"
    aria-describedby="alert-dialog-description"
  >
    {title ? (
      <DialogTitle id="alert-dialog-title">{title}</DialogTitle>
    ) : null}

    <DialogContent>
      <DialogContentText id="alert-dialog-description">
        {description}
      </DialogContentText>
    </DialogContent>
    <DialogActions>
      {hasCloseButton ? (
        <Button onClick={onClose} color="primary">
          {disagreeText}
        </Button>
      ) : null}

      <Button
        onClick={onConfirm}
        color="primary"
        autoFocus
        className={classes.primaryButton}
      >
        {agreeText}
      </Button>
    </DialogActions>
  </Dialog>
)

ConfirmBranch.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  title: PropTypes.string,
  description: PropTypes.string.isRequired,
  agreeText: PropTypes.string.isRequired,
  disagreeText: PropTypes.string.isRequired,
  onClose: PropTypes.func.isRequired,
  onConfirm: PropTypes.func.isRequired,
  hasCloseButton: PropTypes.bool.isRequired,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
}

ConfirmBranch.defaultProps = {
  title: null,
}

export default withStyles(styles)(ConfirmBranch)
