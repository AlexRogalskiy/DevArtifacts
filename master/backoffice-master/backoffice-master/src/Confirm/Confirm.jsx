import React from 'react'
import PropTypes from 'prop-types'

import ConfirmBranch from './ConfirmBranch'

const withConfirm = Component => class extends React.Component {
  static propTypes = {
    isOpen: PropTypes.bool,
    title: PropTypes.string,
    description: PropTypes.string.isRequired,
    agreeText: PropTypes.string,
    disagreeText: PropTypes.string,
    onConfirm: PropTypes.func.isRequired,
    onClose: PropTypes.func,
    hasCloseButton: PropTypes.bool,
  }

  static defaultProps = {
    title: null,
    isOpen: false,
    agreeText: 'Agree',
    disagreeText: 'Disagree',
    onClose: () => {},
    hasCloseButton: true,
  }

  constructor(props) {
    super(props)

    this.state = {
      isOpen: false,
    }

    this.handleClose = this.handleClose.bind(this)
    this.handleConfirm = this.handleConfirm.bind(this)
  }

  componentWillMount() {
    this.setState({
      isOpen: this.props.isOpen,
    })
  }

  componentWillReceiveProps(nextProps) {
    if (this.state.isOpen !== nextProps.isOpen) {
      this.setState({
        isOpen: nextProps.isOpen,
      })
    }
  }

  handleClose() {
    this.setState({
      isOpen: false,
    })

    this.props.onClose()
  }

  handleConfirm() {
    this.setState({
      isOpen: false,
    })

    this.props.onConfirm()
  }

  render() {
    return (
      <Component
        {...this.props}
        {...this.state}
        onClose={this.handleClose}
        onConfirm={this.handleConfirm}
      />
    )
  }
}

export default withConfirm(ConfirmBranch)
