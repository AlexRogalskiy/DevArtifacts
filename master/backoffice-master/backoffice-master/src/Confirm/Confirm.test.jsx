import React from 'react'
import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import Confirm from './'

Enzyme.configure({ adapter: new Adapter() })

describe('Confirm', () => {
  it('renders correctly', () => {
    const tree = mount(<Confirm
      title="Title"
      description="Desc"
      onConfirm={() => { }}
    />)

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly without close button', () => {
    const tree = mount(<Confirm
      title="Title"
      description="Desc"
      onConfirm={() => { }}
      hasCloseButton={false}
    />)

    expect(tree).toMatchSnapshot()
  })

  it('confirm is handled', () => {
    const onConfirm = jest.fn()
    const confirm = shallow((
      <Confirm
        isOpen
        title="Title"
        description="Desc"
        onConfirm={onConfirm}
      />
    ))

    expect(confirm.state().isOpen).toBe(true)
    confirm.instance().handleConfirm()

    expect(onConfirm).toHaveBeenCalled()
  })

  it('close is handled', () => {
    const onClose = jest.fn()
    const confirm = shallow((
      <Confirm
        isOpen
        title="Title"
        description="Desc"
        onConfirm={() => {}}
        onClose={onClose}
      />
    ))

    expect(confirm.state().isOpen).toBe(true)
    confirm.instance().handleClose()

    expect(onClose).toHaveBeenCalled()
  })

  it('changes open', () => {
    const confirm = shallow((
      <Confirm
        isOpen
        description="Desc"
        onConfirm={() => { }}
      />
    ))

    confirm.setProps({
      isOpen: false,
    })
  })
})
