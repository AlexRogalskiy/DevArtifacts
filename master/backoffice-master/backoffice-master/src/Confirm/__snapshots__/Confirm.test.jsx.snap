// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm renders correctly 1`] = `""`;

exports[`Confirm renders correctly without close button 1`] = `""`;
